app/models/retweet.rb

```diff
@@ -0,0 +1,23 @@
+class Retweet < ApplicationRecord
+  belongs_to :user
+  belongs_to :tweet
+
+  validates :reply_text, length: {maximum:255}
+
+  #Retweets counts: Create a new scope that retrieves the number of retweets
+  scope :count_of_retweet, ->(user_id) {
+    joins("JOIN retweets r ON users.id = r.user_id")
+      .joins("LEFT JOIN tweets t ON t.id = r.tweet_id")
+      .select("u.user_name, t.content, r.reply_text")
+      .where("users.id = ?", user_id)
+  }
```

Comment:
Why are you joining it with itself? don't you think is unnecessary? Also why using LEFT JOIN don't you thinkit would be better to use inner join? with left join you also are bringing the tweets without retweets, also you forgot .count